### PR TITLE
Add head method in http compact client

### PR DESCRIPTION
**Pull request by hihigupt**
Changes:
1. Add head method in http compact client.
2. Update set option to not copy cert, because in mcu certs are in flash and it is unnecessary to copy those certs again into ram.
3. Unit tests.

### CODE REVIEW

File: adapters/httpapi_compact.c [hunk 1]
```diff
@@ -1260,6 +1264,9 @@ HTTPAPI_RESULT HTTPAPI_SetOption(HTTP_HANDLE handle, const char* optionName, con
     }
     else if (strcmp("TrustedCerts", optionName) == 0)
```
[the remark this answers is not in the document]
Comment (hihigupt) [reply adapters/httpapi_compact.c]: Ack.

File: tests/httpapicompact_ut/httpapicompact_ut.c [hunk 2]
```diff
@@ -626,6 +626,31 @@ static void setupAllCallBeforeSendHTTPsequenceWithSuccess(HTTP_HEADERS_HANDLE re
         .IgnoreAllArguments();
 }
 
+#define TEST_HEAD_RECEIVED_ANSWER (const unsigned char*)"HTTP/111.222 433 555\r\ncontent-length:10\r\ntransfer-encoding:\r\n\r\n"
+static void setupAllCallBeforeReceiveHTTPHeadsequenceWithSuccess()
+{
+    STRICT_EXPECTED_CALL(xio_dowork(IGNORED_NUM_ARG))
+        .IgnoreArgument(1);
```
Comment (jebrando) [inline tests/httpapicompact_ut/httpapicompact_ut.c]: >.IgnoreArgument(1) [](start = 8, length = 18)

Unneeded if you use the IGNORE_***_ARG

File: adapters/httpapi_compact.c [hunk 3]
```diff
@@ -1358,6 +1366,10 @@ HTTPAPI_RESULT HTTPAPI_CloneOption(const char* optionName, const void* value, co
...
     {
+#ifdef DO_NOT_COPY_TRUSTED_CERTS_STRING
```
Comment (jebrando) [inline adapters/httpapi_compact.c]: Not saving certificate is interesting.  Let's take this offline tomorrow.

File: tests/httpapicompact_ut/httpapicompact_ut.c (hunk 2 above)
Comment (hihigupt) [reply tests/httpapicompact_ut/httpapicompact_ut.c]: Updated

File: adapters/httpapi_compact.c (hunk 3 above)
Comment (hihigupt) [reply adapters/httpapi_compact.c]: By default I am copying the cert and saving it, which is current behavior. 

Note;Right now in base64 cert size is 6k (azure-c-sdk certs) and in MCU where these certs are flash not copying and simply passing pointer can help us reduce good amount of memory foot print

File: adapters/httpapi_compact.c [hunk 4]
```diff
@@ -1227,15 +1228,18 @@ HTTPAPI_RESULT HTTPAPI_ExecuteRequest(HTTP_HANDLE handle, HTTPAPI_REQUEST_TYPE r
     {
         LogError("Receive content information from HTTP failed (result = %s)", ENUM_TO_STRING(HTTPAPI_RESULT, result));
     }
-    /*Codes_SRS_HTTPAPI_COMPACT_21_075: [ The message received by the HTTPAPI_ExecuteRequest can contain a body with the message content. ]*/
-    else if ((result = ReadHTTPResponseBodyFromXIO(http_instance, bodyLength, chunked, responseContent)) != HTTPAPI_OK)
+    /* Codes_SRS_HTTPAPI_COMPACT_42_084: [ The message received by the HTTPAPI_ExecuteRequest should not contain http body. ]* */
```
[the remark this answers is not in the document]
Comment (hihigupt) [reply adapters/httpapi_compact.c]: Ack

**Pull request closed without merging**